iset.mm

```diff
@@ -147954,6 +147981,516 @@ of the multiplicative monoid ( ~ df-mgp ) of a ring-like structure.  This
+      SZADUJZUKZFUFUMYHFUGXBXCYGULUNXAXBFUOTZUHOZXDXEXFUAUIZQZUBUIZQZXDXEYMQZXD
+      XFYMQZYKQZRZXDXEYKQZXFYMQZYPXEXFYMQZYKQZRZSZCUCUIZUJZBUUEUJZUDUIZXDYMQZUU
+      HRZXDUUHYMQZUUHRZSZSZAUUEUJZUDIUPZUBGUPZUAEUPZUCDUPZSZSXBXCYGSZSWTYHXAUUT
+      UVAXBXAYJXCUUSYGYJXCUQXAXCYJHYIUHKURUSUTXAUURYGUCDPXADFVATZPJVAPVBXAUVBPO
+      ZVFUVCPFVAFVAVCVDVGVEXAUUEDRZSZUUQYGUAEPXAEPOUVDXAEFVHTZPLFVHPVIVJVEVKUVE
+      YKERZSZUUPYGUBGPXAGPOUVDUVGXAGFVPTZPMFVPPVLVJVEVMUVHYMGRZSZUUOYGUDIPXAIPO
+      UVDUVGUVJXAIFVNTZPNVNPVBXAUVLPOZVQUVMPFVNFVNVCVDVGVEVOUVKUUHIRZSZUUNYFAUU
+      EDXAUVDUVGUVJUVNVRZUVOUUGXTUUMYEUVOUUFXSBUUEDUVPUVOUUDXRCUUEDUVPUVOYRXLUU
+      CXQUVOYNXHYQXKUVOXDXDYLXGYMGUVHUVJUVNWFZUVOXDVSZUVOYKEXEXFUVEUVGUVJUVNVTZ
+      WAWBUVOYOXIYPXJYKEUVSUVOYMGXDXEUVQWAUVOYMGXDXFUVQWAZWBWCUVOYTXNUUBXPUVOYS
+      XMXFXFYMGUVQUVOYKEXDXEUVSWAUVOXFVSWBUVOYPXJUUAXOYKEUVSUVTUVOYMGXEXFUVQWAW
+      BWCWDWEWEUVOUUJYBUULYDUVOUUIYAUUHIUVOUUHIXDXDYMGUVQUVKUVNWPZUVRWBUWAWCUVO
+      UUKYCUUHIUVOXDXDUUHIYMGUVQUVRUWAWBUWAWCWDWDWEWGWGWGWGWDWHUEUIZUOTZUHOZUUO
+      UDUWBVNTZUPZUBUWBVPTZUPZUAUWBVHTZUPZUCUWBVATZUPZSUUTUEFUGUFUWBFRZUWDYJUWL
+      UUSUWMUWCYIUHUWBFUOWQWIUWMUWJUURUCUWKDUWMUWKUVBDUWBFVAWQJWJUWMUWHUUQUAUWI
+      EUWMUWIUVFEUWBFVHWQLWJUWMUWFUUPUBUWGGUWMUWGUVIGUWBFVPWQMWJUWMUUOUDUWEIUWM
+      UWEUVLIUWBFVNWQNWJWKWLWLWLWDABCUBUEUDUCUAWMWNXBXCYGWOWRWS $.
+  $}
+
+  ${
+    $d x y z R $.
+    $( A semiring is a commutative monoid.  (Contributed by Thierry Arnoux,
+       21-Mar-2018.) $)
+    srgcmn $p |- ( R e. SRing -> R e. CMnd ) $=
+      ( vx vy vz csrg wcel ccmn cmgp cfv cmnd cv cplusg cmulr wceq cbs wral c0g
+      co wa eqid issrg simp1bi ) AEFAGFAHIZJFBKZCKZDKZALIZRAMIZRUDUEUHRUDUFUHRZ
+      UGRNUDUEUGRUFUHRUIUEUFUHRUGRNSDAOIZPCUJPAQIZUDUHRUKNUDUKUHRUKNSSBUJPBCDUJ
+      UGAUHUCUKUJTUCTUGTUHTUKTUAUB $.
+
+    $( A semiring is a monoid.  (Contributed by Thierry Arnoux,
+       21-Mar-2018.) $)
+    srgmnd $p |- ( R e. SRing -> R e. Mnd ) $=
+      ( csrg wcel ccmn cmnd srgcmn cmnmnd syl ) ABCADCAECAFAGH $.
+
+    srgmgp.g $e |- G = ( mulGrp ` R ) $.
+    $( A semiring is a monoid under multiplication.  (Contributed by Thierry
+       Arnoux, 21-Mar-2018.) $)
+    srgmgp $p |- ( R e. SRing -> G e. Mnd ) $=
+      ( vx vy vz csrg wcel ccmn cmnd cv cplusg cfv co cmulr wceq cbs wral eqid
+      wa c0g issrg simp2bi ) AGHAIHBJHDKZEKZFKZALMZNAOMZNUDUEUHNUDUFUHNZUGNPUDU
+      EUGNUFUHNUIUEUFUHNUGNPTFAQMZREUJRAUAMZUDUHNUKPUDUKUHNUKPTTDUJRDEFUJUGAUHB
+      UKUJSCUGSUHSUKSUBUC $.
+  $}
+
+  ${
+    $d x y z B $.  $d x y z R $.  $d x y z .x. $.  $d x y z X $.  $d x y z Y $.
+    $d x y z .+ $.  $d x y z Z $.
+    srgi.b $e |- B = ( Base ` R ) $.
+    srgi.p $e |- .+ = ( +g ` R ) $.
+    srgi.t $e |- .x. = ( .r ` R ) $.
+    $( Properties of a semiring.  (Contributed by NM, 26-Aug-2011.)  (Revised
+       by Mario Carneiro, 6-Jan-2015.)  (Revised by Thierry Arnoux,
+       1-Apr-2018.) $)
+    srgi $p |- ( ( R e. SRing /\ ( X e. B /\ Y e. B /\ Z e. B ) )
```

Comment:
This is the distributive law, isn't it? Then the label should be changed to `srgdi` (see ~ringdi and ~ringdir for rings), and the comment should be "Distributive law for the multiplication operation of a seimiring (left- and right-distributivity)." (as for rings).

Reply:
Sorry, we have aready ~srgdi (and ~srgdir), see below. And we have an alogous theorem for rings (~ringi). 

Although I do not know why we need this combinded theorems for the distributive law (they are used in the two separate theorems only, for semirings as well as for rings), it should be OK for now.

Reply:
Maybe it can be made a lemma (like ~srgidmlem).